### ISSUE TITLE
flag aria-modal as not accessibility supported

### ISSUE THREAD

**Issue by dylanb** [dequelabs/axe-core#3428]
Product: axe-core

Expectation: aria-modal is not supported by Firefox or IE https://caniuse.com/?search=aria-modal and https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaModal and should therefore be flagged by axe-core

Actual: aria-modal is not flagged

Motivation: This adheres to the axe-core principles for accessibility supported https://github.com/dequelabs/axe-core/blob/develop/doc/accessibility-supported.md#impact-on-aria

**Comment by WilcoFiers** [comment]
The bar for axe-core flagging ARIA attributes as violations is very high. I don't think `aria-modal` qualifies here. Axe-core only fail things that **causes** accessibility problems in **all plausible scenarios**. Here's what that policy has to say about it:

> If the feature's use does not have a negative impact on accessibility: allow

I can't think of an example in which removing aria-model would make an accessibility page inaccessible. I can't imagine that would be the case for **all** pages that had `aria-modal`. It's not like aria-description, where it can be used to communicate vital information.  `aria-modal` is essentially just a hook for screen readers to trigger some optional behaviour.  It does nice things, but nothing critical to accessibility as far as i can see.

> Assistive technologies MAY limit navigation to the modal element's contents

Ergo; if you need the rest of the content hidden, unlike what MDN says you will need to use `aria-hidden`.

> When a modal element is displayed, assistive technologies SHOULD navigate to the element

That's nice and all, but not something content authors can rely on. Keyboard users need that focus moved too. `aria-modal` is not an alternative to focus management.

> When a modal element is displayed, authors SHOULD mark all other contents as inert

Yet another thing content authors can not trust `aria-modal` to do.

**Comment by WilcoFiers** [comment]
From the e-mail I got the impression you agree @dylanb, so I'm going to close this.

**Comment by melaniephilipp** [comment]
@WilcoFiers I have a half-written comment on this. But we can talk about it outside of this forum.
